editor/components/table-of-contents/index.js

```diff
@@ -34,49 +23,16 @@ function TableOfContents( { blocks } ) {
 					icon="info-outline"
 					aria-expanded={ isOpen }
 					label={ __( 'Content Structure' ) }
-					disabled={ blocks.length === 0 }
+					disabled={ ! hasBlocks }
 				/>
 			) }
-			renderContent={ () => ( [
-				<div key="counts" className="table-of-contents__counts">
-					<div className="table-of-contents__count">
-						{ __( 'Words' ) }
-						<WordCount />
-					</div>
-					<div className="table-of-contents__count">
-						{ __( 'Headings' ) }
-						<span className="table-of-contents__number">{ headings.length }</span>
-					</div>
-					<div className="table-of-contents__count">
-						{ __( 'Paragraphs' ) }
-						<span className="table-of-contents__number">{ paragraphs.length }</span>
-					</div>
-					<div className="table-of-contents__count">
-						{ __( 'Blocks' ) }
-						<span className="table-of-contents__number">{ blocks.length }</span>
-					</div>
-				</div>,
-				headings.length > 0 && (
-					<div key="headings">
-						<hr />
-						<span className="table-of-contents__title">{ __( 'Document Outline' ) }</span>
-						<DocumentOutline />
-					</div>
-				),
-			] ) }
+			renderContent={ TableOfContentsPanel }
```

Comment:
That's interesting. I assumed you always need to pass function here 👍

Reply:
```js
{ renderContent( args ) }
```
This is how it is rendered later. @youknowriad can you confirm that we can pass component here?

Reply:
It's meant to be a function yes, we could definitely update it to support this?

Reply:
It works properly because both `Query` HOC and `TableOfContentsPanel` are built as functional components.

Reply:
This is what happens when you pass a class component as `renderContent`:
![screen shot 2018-02-14 at 11 19 37](https://user-images.githubusercontent.com/699132/36199162-fb484738-1178-11e8-9ee3-1c526385b3bf.png)